### PR TITLE
feat: move autocomplete suggestions calculation to web worker

### PR DESCRIPTION
**Pull request by mstykow**
### Summary of changes

- adjust CSP to allow web worker blobs
- create signals web worker with hydration mechanism for expensive calculations

NOTE: There will be two more PRs to refactor the other existing web workers with this new approach in order to reduce the memory footprint on the application.

### Context and reason for change

#2384 

### How can the changes be tested

Check that going between resources in the audit view's resource browser is not feeling slow.

### CODE REVIEW

**Review by MarkusObendrauf — DISMISSED** [review]
I'm getting out of memory errors on production scans after this worker is introduced. We were already teetering close to the 4GB limit and this is a step too far.

If I wait a while after loading the file I am sometimes able to click on a resource to load the worker, so I was able to load a profile. The new worker really is taking up a lot of memory.
image

**Comment by MarkusObendrauf** [comment]
(code review is incomplete as I noticed the OOM issues)

File: src/Frontend/state/actions/view-actions/view-actions.ts [hunk 1]
```diff
@@ -59,7 +59,7 @@ export function navigateToView(view: View): AppThunkAction {
   };
 }
 
-function setView(view: View): SetView {
+export function setView(view: View): SetView {
```
Comment (alexzurbonsen) [inline src/Frontend/state/actions/view-actions/view-actions.ts:62]: I am not sure it is a good idea to export this function? There is aslo `navigateToView` which does more then setting the view and is probably the correct way to switch views. Exporting `setView` might lead to incorrect usages in the future?

**Review by alexzurbonsen — APPROVED** [review]

File: src/Frontend/util/use-variable.ts [hunk 2]
```diff
@@ -17,8 +18,9 @@ import { State } from '../types/types';
 export function useVariable<T>(
   name: string,
   initialValue: T,
-): [T, (newValue: T) => void] {
+): [T, (value: T | ((prev: T) => T)) => void] {
```
[the remark this answers is not in the document]
Comment (mstykow) [reply src/Frontend/util/use-variable.ts:21]: not needed

**Pull request merged**